Library/Homebrew/cmd/vendor-install.sh

```diff
@@ -4,42 +4,49 @@
 #:  Install Homebrew's portable Ruby.
 
 # Don't need shellcheck to follow this `source`.
-# shellcheck disable=SC1090
-source "$HOMEBREW_LIBRARY/Homebrew/utils/lock.sh"
+# set from user environment
+# shellcheck disable=SC1090,SC2154
+source "${HOMEBREW_LIBRARY}/Homebrew/utils/lock.sh"
 
-VENDOR_DIR="$HOMEBREW_LIBRARY/Homebrew/vendor"
+VENDOR_DIR="${HOMEBREW_LIBRARY}/Homebrew/vendor"
 
 # Built from https://github.com/Homebrew/homebrew-portable-ruby.
 #
 # Dynamic variables can't be detected by shellcheck
-# shellcheck disable=SC2034
-if [[ -n "$HOMEBREW_MACOS" ]]
+# set from user environment
```

Comment:
This (and most others) isn't set from user environment; can you check the definitions of these?